patches/build/git.20231122.968682938b.diff

```diff
@@ -0,0 +1,151 @@
+   ]
+ 
+   cflags_cc = [
+   ]
+ 
+   asmflags = [
+-    "-Wa,GOFF",
++    "-Wa,-mGOFF",
++    "-fno-integrated-as",
+   ]
+ 
+   ldflags = [
+diff --git a/toolchain/toolchain.gni b/toolchain/toolchain.gni
+index 23ec903ad..734ba8a7c 100644
+--- a/toolchain/toolchain.gni
++++ b/toolchain/toolchain.gni
+@@ -72,7 +72,7 @@ if (host_os == "mac") {
+   host_shlib_extension = ".dylib"
+ } else if (host_os == "win") {
+   host_shlib_extension = ".dll"
+-} else if (host_os == "linux" || host_os == "aix") {
++} else if (host_os == "linux" || host_os == "aix" || host_os == "zos") {
+   host_shlib_extension = ".so"
+ } else {
+   assert(false, "Host platform not supported")
+diff --git a/toolchain/zos/BUILD.gn b/toolchain/zos/BUILD.gn
+index 3af5f8033..20eef2caf 100644
+--- a/toolchain/zos/BUILD.gn
++++ b/toolchain/zos/BUILD.gn
+@@ -7,9 +7,9 @@
+ import("//build/toolchain/gcc_toolchain.gni")
+ 
+ toolchain("s390x") {
+-  cc = "xlclang"
+-  cxx = "xlclang++"
+-  asm = "xlclang"
++  cc = "clang"
++  cxx = "clang++"
++  asm = "clang"
+   ar = "ar"
+   ld = cxx
+ 
+@@ -36,7 +36,7 @@ toolchain("s390x") {
+ 
+   tool("cc") {
+     depfile = "{{output}}.d"
+-    command = "$cc -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{cflags}} {{cflags_c}}${extra_cflags} -c {{source}} -o {{output}}"
++    command = "$cc -MMD -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{cflags}} {{cflags_c}}${extra_cflags} -c {{source}} -o {{output}}"
+     depsformat = "gcc"
+     description = "CC {{output}}"
+     outputs = [ "$object_subdir/{{source_name_part}}.o" ]
+@@ -44,7 +44,7 @@ toolchain("s390x") {
+ 
+   tool("cxx") {
+     depfile = "{{output}}.d"
+-    command = "$cxx -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}}${extra_cppflags}${extra_cxxflags} -c {{source}} -o {{output}}"
++    command = "$cxx -MMD -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}}${extra_cppflags}${extra_cxxflags} -c {{source}} -o {{output}}"
+     depsformat = "gcc"
+     description = "CXX {{output}}"
+     outputs = [ "$object_subdir/{{source_name_part}}.o" ]
+@@ -53,7 +53,7 @@ toolchain("s390x") {
+   tool("asm") {
+     # Just use the C compiler to compile assembly.
+     depfile = "{{output}}.d"
+-    command = "$asm -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{asmflags}}${extra_asmflags} -c {{source}} -o {{output}}"
++    command = "$asm ${rebuild_string} {{asmflags}}${extra_asmflags} -c {{source}} -o {{output}}"
+     depsformat = "gcc"
+     description = "ASM {{output}}"
+     outputs = [ "$object_subdir/{{source_name_part}}.o" ]
+@@ -88,7 +88,7 @@ toolchain("s390x") {
+     # implement (1) linking to produce a .so, (2) extracting the symbols
+     # from that file (3) if the extracted list differs from the existing
+     # .TOC file, overwrite it, otherwise, don't change it.
+-    link_command = "$ld -Wl,DLL {{ldflags}}${extra_ldflags} -o \"$sofile\" `cat $rspfile`"
```

Comment:
will this break Linux on Z?

Reply:
I have no clue. I’m on holiday.  Do we wish to merge today or wait until Monday?

Reply:
No, this file is a z/OS-only file `toolchain/zos/BUILD.gn` (line 99).

Reply:
@netskink I'm hoping Igor will have a chance to review it first before merging.